src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -923,42 +923,69 @@ def update(
         assignments: Dict[str, str],
         condition: Optional[str],
         source_data: Optional[SnowflakePlan],
+        source_plan: Optional[LogicalPlan],
     ) -> SnowflakePlan:
-        return self.query(
-            update_statement(
-                table_name,
-                assignments,
-                condition,
-                source_data.queries[-1].sql
-                if source_data and source_data.queries
-                else None,
-            ),
-            None,
-        )
+        if source_data:
+            return self.build(
+                lambda x: update_statement(
```

Comment:
I'm wondering if it is possible to support `None` child in `build` that delegates to `query`? Is there some gotcha that I might be missing?

Reply:
No, `build` requires a valid `child` https://github.com/snowflakedb/snowpark-python/blob/96e96a0fb431a32278d8a9463cd4ba0f1c28b91d/src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py#L246

Reply:
Can we do the `if` check similar to what we do here but hide it in `build`?

Reply:
hmm I'm not sure - many methods already use `build` so it's a little risky, and we can do it later.